rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/filter/EndpointsFilter.java

```diff
@@ -19,25 +31,75 @@
 @Component
 public class EndpointsFilter extends OncePerRequestFilter {
 
-    private static final String ERROR_MESSAGE = "Service is disabled";
-
-    private static final String CREATE_USER_ENDPOINT = "/resource/api/users";
-
-    @Value("${sw360.rest.api.createuser.disabled}")
-    boolean disabledUsrCreation;
+    @Value("${blacklist.sw360.rest.api.endpoints}")
+    String endpointsTobeBlackListed;
 
     @Override
     protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
             throws ServletException, IOException {
+
         String requestURI = request.getRequestURI();
         String method = request.getMethod();
 
-        if (disabledUsrCreation && requestURI.equalsIgnoreCase(CREATE_USER_ENDPOINT) && method.equals("POST")) {
-            response.sendError(HttpStatus.SERVICE_UNAVAILABLE.value(), ERROR_MESSAGE);
-        } else {
+        String[] endpointMethodPairs = endpointsTobeBlackListed.split(",");
+        Map<String, Set<String>> endpointHttpMethods = getMapOfEndpointToHttpMethods(endpointMethodPairs);
+        boolean isAMatch = verifyMatchingOfRequestURIToEndpoints(requestURI, endpointHttpMethods);
+
+        if (!isAMatch) {
             filterChain.doFilter(request, response);
+        } else {
+            Set<String> httpMethodsToBeBlocked = new HashSet<>();
+            Optional<Entry<String, Set<String>>> vv = endpointHttpMethods.entrySet().stream().filter(es -> {
```

Comment:
Please use proper variable names for vv, ep.

Reply:
Done